x/zkauth/ante/zkauth.go

```diff
@@ -114,6 +115,56 @@ func (zkisd ZKAuthIncrementSequenceDecorator) AnteHandle(ctx sdk.Context, tx sdk
 	return next(ctx, tx, simulate)
 }
 
+type ZKAuthDeductFeeDecorator struct {
+	ak             authante.AccountKeeper
+	bankKeeper     authtypes.BankKeeper
+	feegrantKeeper authante.FeegrantKeeper
+}
+
+func NewZKAuthDeductFeeDecorator(ak authante.AccountKeeper, bankKeeper authtypes.BankKeeper, feegrantKeeper authante.FeegrantKeeper) ZKAuthDeductFeeDecorator {
+	return ZKAuthDeductFeeDecorator{ak: ak, bankKeeper: bankKeeper, feegrantKeeper: feegrantKeeper}
+}
+
+func (zdf ZKAuthDeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	isZKAuthTx, _, _, err := getZKAuthInfoFromTx(tx)
+	if err != nil {
+		return ctx, err
+	}
+
+	if !isZKAuthTx {
+		dfd := authante.NewDeductFeeDecorator(zdf.ak, zdf.bankKeeper, zdf.feegrantKeeper)
+		return dfd.AnteHandle(ctx, tx, simulate, next)
+	}
```

Comment:
Rather than creating `ante.NewDeductFeeDecorator` every time, why not use the antihandler parameter to replace when creating the antiHandler, like this:
```golang
zkauthante.NewZKAuthDeductFeeDecorator(opts.AccountKeeper, opts.BankKeeper, opts.FeegrantKeeper, 
     ante.NewDeductFeeDecorator(opts.AccountKeeper, opts.BankKeeper, opts.FeegrantKeeper)
),
```